### PR TITLE
DEV-1125: duplicate holdings cleanup

### PR DESCRIPTION
**Pull request by aelkiss**
## Context

Many clusters have duplicate holdings for reasons that have previously been discussed; see DEV-1125

## Description

This cleans up those clusters by iterating through them one by one, grouping them by update key the same way that `find_old_holdings` does (https://github.com/hathitrust/holdings-backend/blob/main/lib/clustering/cluster_holding.rb#L86-L91) and retaining the most recent in each equality group.

@mwarin Mainly I'm interested if you can think of any other tests we should be doing to make sure this does what we want.

This can be reviewed now; once approved we should wait to merge it until after #301

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ bash bin/setup/setup_dev.sh
 
 ## Running the tests
 
-`docker-compose run --rm dev bundle exec rspec`
+`docker-compose run --rm test`
```
Comment (mwarin) [inline README.md:21]: might as well make it hyphenless: `docker compose`

File: bin/setup/setup_dev.sh [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
 $(dirname ${BASH_SOURCE[0]})/setup_test.sh
 
-docker compose up -d mongo_dev pushgateway redis
-docker compose run --rm dev bin/setup/wait-for mongo_dev:27017 -- echo "mongo is ready"
+docker compose up --wait mongo_dev
 docker compose exec -T mongo_dev bash /tmp/bin/setup/rs_initiate.sh mongo_dev
-docker compose run --rm -e MONGOID_ENV=development dev bundle exec ruby lib/tasks/build_database.rb
```
Comment (mwarin) [inline bin/setup/setup_dev.sh:8]: What is the context for doing this? 
This feels more functional than a tidying, and deserves to be mentioned, no?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,74 +1,100 @@
...
       - ./bin:/tmp/bin
+    healthcheck: *mongo-healthcheck
 
   mariadb:
     image: mariadb
     environment:
       MYSQL_RANDOM_ROOT_PASSWORD: 1
     volumes:
       - ./sql:/docker-entrypoint-initdb.d/
+    healthcheck:
```
Comment (mwarin) [inline docker-compose.yml:78]: This stuff I also wonder, did you mean to put in a different PR?

File: spec/cleanup_duplicate_holdings_spec.rb [hunk 4]
```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
```
Comment (mwarin) [inline spec/cleanup_duplicate_holdings_spec.rb:1]: `bundle exec rspec spec/cleanup_duplicate_holdings_spec.rb` runs well.
I don't see any missing tests.

**Review by mwarin — APPROVED** [review]
I'm mostly on board with the functional changes, but I don't understand how you would execute the script as written.

I was confused seeing some of (all?) the small unrelated (?) changes I already approved in PR #300 . Could those have been rebased away?

Not going to let my confusion stand in the way of an APPROVE, though.

**Comment by aelkiss** [comment]
Right, this needs to be rebased...

File: bin/cleanup_duplicate_holdings.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply bin/cleanup_duplicate_holdings.rb:1]: Right, there's probably some missing stuff here that the specs hide. I'll address that and add a comment as to the purpose here.

File: bin/setup/setup_dev.sh (hunk 2 above)
Comment (aelkiss) [reply bin/setup/setup_dev.sh:8]: See #301

**Comment by aelkiss** [comment]
I should have specified that only https://github.com/hathitrust/holdings-backend/pull/302/commits/90ccd6650e96d1a4bb32a4efd2a9cbea72c818b3 needed to be reviewed and the other changes were covered in #300 or #301

**Comment by aelkiss** [comment]
Rebased against main & added some fixes in the bin script for cleaning up duplicate holdings. Will merge after tests pass.
